### PR TITLE
Axpd-7704: Support Data Sources

### PR DESCRIPTION
**Pull request by danny-akp**
We are supporting data sources for the following resources:

- axual_group
- axual_environment
- axual_topic
- axual_application
- axual_application_access_grant
- axual_schema_version

A documentiation on how to use data sources

### CODE REVIEW

File: axual-webclient/application_access_grant.go [hunk 1]
```diff
@@ -75,3 +76,30 @@ func (c *Client) RevokeOrDenyGrant(applicationAccessGrantId string, reason strin
...
+	headers := map[string]string{
+		"Content-Type": "application/json",
+		"Accept":       "application/hal+json",
+	}
+
+	values := url.Values{}
+	values.Add("streamId", data.TopicId)
+	values.Add("applicationId", data.ApplicationId)
+	values.Add("environmentId", data.EnvironmentId)
+	values.Add("accessType", data.AccessType)
```
Comment (markaxual) [inline axual-webclient/application_access_grant.go:97]: This function should be generic enough to handle all valid scenarios of `GetApplicationAccessGrantsByAttributes` so all these attributes must be optional, and all the other possible attributes should be possible.

File: axual-webclient/environment.go [hunk 2]
```diff
@@ -50,11 +51,20 @@ func (c *Client) DeleteEnvironment(id string) error {
 	return nil
 }
 
-func (c *Client) ReadEnvironments() (*EnvironmentsResponse, error) {
-	o := EnvironmentsResponse{}
+func (c *Client) ReadEnvironments() (*EnvironmentByNameResponse, error) {
+	o := EnvironmentByNameResponse{}
```
Comment (markaxual) [inline axual-webclient/environment.go:55]: Why are we changing the response here?

File: axual-webclient/groups_data.go [hunk 3]
```diff
@@ -119,3 +119,52 @@ type GroupRequest struct {
 	PhoneNumber  interface{} `json:"phoneNumber,omitempty"`
 	Members      []string    `json:"members,omitempty"`
 }
+
+type GroupByNameResponse struct {
```
Comment (markaxual) [inline axual-webclient/groups_data.go:123]: Group what by name? It's a bit ambiguous. Maybe `GetGroupsByNameResponse`

**Review by markaxual — CHANGES_REQUESTED** [review]
Nice work. Some comments

**Comment by markaxual** [comment]
@danny-akp Please update the Changelog

**Review by markaxual — APPROVED** [review]

**Review by yunusaxual — APPROVED** [review]
We tested. It works great.

**Pull request merged**